### DISCUSSION TITLE
Docker Open Web UI not accessible on WSL 2 Ubuntu + Docker Desktop Setup

### DISCUSSION THREAD

**Discussion by Evolutionx86** [open-webui/open-webui#3961] [General]
**Issue: Docker Open Web UI not accessible on WSL 2 Ubuntu + Docker Desktop Setup**

**Description**
================

Given:
Linux Ubuntu over WSL 2 on Windows machine
Docker Desktop

The official command to create a Docker container was run successfully, but the Open Web UI was not accessible.

```bash
docker run -d --network=host -v open-webui:/app/backend/data -e OLLAMA_BASE_URL=http://127.0.0.1:11434 --name open-webui --restart always ghcr.io/open-webui/open-webui:main
```

http://127.0.0.1:8080/ --> not accessible

**Workaround**
=============

Using port mapping, the Open Web UI became accessible on Port 8080 from the host system.

```bash
docker run -d -p 8080:8080 -v open-webui:/app/backend/data -e OLLAMA_BASE_URL=http://127.0.0.1:11434 --name open-webui --restart always ghcr.io/open-webui/open-webui:main
```

http://127.0.0.1:8080/ --> accessible

**Problem**
==========

However, the Docker container was unable to communicate with the Ollama on the host system.

```bash
http://localhost:11434/ -> Ollama is running
```

But the Docker container cannot communicate.

**Solution**
==========

Enabling Host Networking in Docker Desktop resolves this issue. To do so:

1. Navigate to the Features in development tab in Settings.
2. Select Enable host networking.

![Docker Enable host networking](https://github.com/user-attachments/assets/3ebeea94-cc37-4c50-89d9-2000134eade6)

**Reproduces**
================

Yes, this issue is reproducible on WSL 2 Ubuntu + Docker Desktop setup.

**Steps to Reproduce**
=====================

1. Install WSL 2 and Docker Desktop.
2. Run the Docker container with the official command.
3. Verify that the Open Web UI is not accessible.
4. Apply the workaround by using port mapping.
5. Verify that the Open Web UI is accessible on Port 8080 from the host system.
6. Try to communicate with Ollama from the Docker container, which should fail.

**Related Issues**
==================

None known at this time.

**PS**

this post was written by my runnig  Llama3 Model on Openweb UI :)

**Comment by drosanda** [comment]
I have same issue with you last night.

I can solved that issue by installing docker service in WSL not in Windows (docker dekstop).
Then make sure the Docker dekstop need to be stopped before start (if any).

**Comment by mxmp210** [comment]
Not a bug, WSL2 creates it's own network subnet by default. You have to explicitly tell it to route traffic between container to windows host by setting NAT rules which is done by given settings under the hood when you check enable host networking in GUI.

Port translation does same, but routes different ports and network in this case. WSL2 runs it's own kernel so NT kernel and linix has to talk to each other via internal layers and thus is by design.

Docker for windows spins another kernel for docker or uses existing linux distribution which can be set in wsl2 integration settings. These settings are wsl2 aware and thus does work while using docker from Windows. It won't work if you spin docker inside Linux as it only sees it's own network and has no config to route traffic between two subnets.

Again this is not a bug, this is how wsl operates at it's core so two kernels can co-exist. Tradeoffs are usually hidden from users.

**Comment by tedlin0124** [comment]
You can use docker-compose file to communicate multiple docker containers

**Comment by geekloper** [comment]
Found an easy way to do it - simply use the IP address of WSL as the Ollama base URL. 
You can obtain this IP address by running the `ipconfig` command. ;)

**Comment by TZECHIN6** [comment]
Hi sir, I am running docker desktop to start the open-webui app, as you mentioned I cannot login to the localhost:8080, even --network=host. How can I solve this issue using docker desktop instead of WSL2.

**Comment by DavidTheBugWriter** [comment]
As an update for those getting here via Google - the "Enable Host Networking" has moved from the image Evolutionx86 posted. It is currently (Jan 25) under settings->Resources-Network:

<img width="863" alt="Screenshot 2025-01-29 at 19 32 30" src="https://github.com/user-attachments/assets/cffb2fce-4b0e-4046-81a9-1d55f2540904" />

**Reply by Evolutionx86** [reply]
Thanks for the update!
